autometa/common/metagenome.py

```diff
@@ -537,88 +595,140 @@ def write_ranks(self, rank='superkingdom'):
         if rank not in NCBI.CANONICAL_RANKS:
             raise ValueError(f'rank: {rank} not in {NCBI.CANONICAL_RANKS}')
         fpaths = []
-        for rank_name,dff in self.taxonomy.groupby(rank):
-            records = [r for r in SeqIO.parse(self.assembly, 'fasta') if r.id in dff.index]
+        for rank_name, dff in self.taxonomy.groupby(rank):
+            records = [r for r in self.seqrecords if r.id in dff.index]
             rank_name = rank_name.replace(' ','_')
             rank_name_fname = '.'.join([rank_name.title(),'fna'])
-            rank_name_fp = os.path.join(self.outdir, rank_name_fname)
+            rank_name_fpath = os.path.join(self.outdir, rank_name_fname)
             if not records:
-                logger.warning(f'No records to write to {rank_name_fp}')
+                logger.warning(f'No records to write to {rank_name_fpath}')
             else:
-                n_written = SeqIO.write(records, rank_name_fp, 'fasta')
-                logger.debug(f'Wrote {n_written} records to {rank_name_fp}')
-                fpaths.append(rank_name_fp)
+                n_written = SeqIO.write(records, rank_name_fpath, 'fasta')
+                logger.debug(f'Wrote {n_written} records to {rank_name_fpath}')
+                fpaths.append(rank_name_fpath)
         return fpaths
 
 def main():
     import argparse
     import logging as logger
     logger.basicConfig(
-        format='%(asctime)s : %(name)s : %(levelname)s : %(message)s',
+        format='[%(asctime)s %(levelname)s] %(name)s: %(message)s',
         datefmt='%m/%d/%Y %I:%M:%S %p',
         level=logger.DEBUG)
+
     parser = argparse.ArgumentParser(description="""
-    Metagenome class holding attributes and methods to manipulate metagenome assemblies.
-    """)
-    parser.add_argument('assembly', help='</path/to/assembly.fasta>')
-    parser.add_argument('--ncbi', help='</path/to/ncbi/dir>', required=True)
-    parser.add_argument('--cutoff', help='length to filter sequences',default=3000,
-        type=int)
-    parser.add_argument('--kmer-size', default=5, type=int)
-    parser.add_argument(
-        '--kmer-normalized',
-        help='Perform CLR transform on k-mer frequencies if provided. (</path/to/kmers.normalized.tsv>)',
-    )
-    parser.add_argument(
-        '--kmer-fpath',
-        help='</path/to/kmers.tsv>',
-    )
+    This handles filtering by length and taxon as well as ORF calling,
+    and various metagenome statistics.
+    """,
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter)
```

Comment:
Do we want to use this formatter in the `template.py` script?

Reply:
This is only required in scripts where we have defaults for parameters specified within parser parameters. Otherwise this is unnecessary. Kind of a judgement call here. We could add it in for easy reference?

Reply:
Is there a consequence to having this formatter when we don't specify defaults? Because I feel if there is no consequence, then adding it might help remove one gotcha.

Reply:
I'll add it to the template. As far as I know, there is not really any consequence to having it as the default formatter.

Reply:
The only consequence I can tell is the help text becomes a bit more verbose with optional parameters that do not have a default resulting in `(default: None)`.

e.g.

```shell
python docs/template.py -h
usage: template.py [-h] [--optional OPTIONAL] [--hello-world HELLO_WORLD]
                   positional

Concise description of script.

positional arguments:
  positional            <help text of positional arg>

optional arguments:
  -h, --help            show this help message and exit
  --optional OPTIONAL   <help text of optional arg> (default: None)
  --hello-world HELLO_WORLD
                        <help text of hello world> (default: Hello World)
```

Reply:
Well, I will leave it up to you. Perhaps it would be nice to include a comment in the template about the options?

Reply:
That's a good point. I'll put it in and comment on the subtle difference.